### ISSUE TITLE
Firebase Database on Android queries with listener does not appear to behave as in iOS

### ISSUE THREAD

**Issue by KevinTheGray** [flutter/flutter#10900]
## Steps to Reproduce

- Use Android, use firebase-database 0.0.9
- Try a query and listener like this
```
// Listen for children added AFTER right now
Query newItemsAddedQuery = itemsReference
              .orderByChild("published-date")
              .startAt(nowInUTCMillis, key: mostRecentItem.pushKey);
          print(mostRecentItem.subtext);
          newItemsAddedQuery.onChildAdded.listen(...
```

On iOS, it will listen for new items added after the specified published date, and the event only fires in that case, which is the behavior I expect.  On Android, the onChildAdded event gets hit for every item it has, and it does not fire when a new item is added.

## Flutter Doctor

[✓] Flutter (on Mac OS X 10.12.4 16E195, locale en-US, channel master)
    • Flutter at /Users/KG/Developer/Flutter/flutter
    • Framework revision 2052741ab3 (29 hours ago), 2017-06-20 11:32:40 -0700
    • Engine revision fffe502d43
    • Tools Dart version 1.24.0-dev.6.7

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.0)
    • Android SDK at /Users/KG/Library/Android/sdk
    • Platform android-26, build-tools 26.0.0
    • ANDROID_HOME = /Users/KG/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Gradle version 3.2
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.1.4)
    • Flutter plugin version 14.0
    • Dart plugin version 171.4694.29

**Comment by eseidelGoogle** [comment]
@collinjackson is likely the right person to see this (and likely already has).

**Comment by jason-simmons** [comment]
I noticed that the handler for startAt in FirebaseDatabasePlugin.java is checking for Integer or Double objects, but not Long or BigInteger types.  nowInUTCMillis probably exceeds the bounds of a Java int, so the StandardMessageCodec would marshal it as one of the larger integer types.

If FirebaseDatabasePlugin doesn't recognize the startAt type, then it will not apply the startAt filter to the query.

Incidentally, the startAt handler is also calling query.endAt instead of query.startAt for Double values.  So changing nowInUTCMillis to a double currently won't work either.

**Comment by collinjackson** [comment]
Thanks for looking into this @jason-simmons. I posted a PR for your review at flutter/plugins#149

**Comment by collinjackson** [comment]
Fixed in 0.0.10

**Comment by kroikie** [comment]
@KevinTheGray

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/989. Any further collaboration will be done there.
